uavcan/file/409.Write.1.0.uavcan

```diff
@@ -0,0 +1,26 @@
+#
+# Write into a remote file.
+# The server shall place the contents of the field 'data' into the file pointed by 'path' at the offset specified by
+# the field 'offset'.
+#
+# When writing a file, the client should repeatedly call this service with data while advancing the offset until the
+# file is written completely. When the write sequence is completed, the client shall call the service one last time,
+# with the offset set to the size of the file and with the data field empty, which will signal the server that the
+# transfer is finished.
+#
+# When the write operation is complete, the server shall truncate the resulting file past the specified offset.
+#
+# Server implementation advice:
+# It is recommended to implement proper handling of concurrent writes to the same file from different clients, for
+# example by maintaining a staging area for uncompleted writes (like FTP servers do).
+#
+
+truncated uint40 offset
+
+Path.1 path
+
+uint8[<=128] data
```

Comment:
Àny reason this is asymmetrical to `Read`?

Reply:
Yes, because we have to transfer both the name and the payload. Having a 256-byte buffer here would result in a structure over 416 bytes large, so I had to pick the next lower power of two. Power of two is important because it is convenient for applications involved with low-level memory access, such as bootloaders.